kartothek/io_components/metapartition.py

```diff
@@ -1550,6 +1553,11 @@ def parse_input_to_metapartition(obj, metadata_version=None):
     obj : Union[Dict, pd.DataFrame, kartothek.io_components.metapartition.MetaPartition]
     metadata_version : int, optional
         The kartothek dataset specification version
+    expected_secondary_indices : Union[Iterable[str], Literal[False], None], optional
```

Comment:
please leave out the optional part. It is already clear from the union type. You could also use `Optional[Union[Iterable[str], Literal[False]]]` if you prefer.